Sources/StreamChat/Workers/Background/ActiveLiveLocationsEndTimeTracker.swift

```diff
@@ -0,0 +1,92 @@
+//
+// Copyright © 2025 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+/// An observer type that observes all active live locations in the database.
+typealias ActiveLiveLocationsObserver = StateLayerDatabaseObserver<ListResult, MessageDTO, MessageDTO>
+
+/// A worker that is responsible for tracking when the end time of active locations is reached.
+class ActiveLiveLocationsEndTimeTracker: Worker {
+    private let activeLiveLocationsObserver: ActiveLiveLocationsObserver
+    private var workItems: [String: DispatchWorkItem] = [:]
```

Comment:
should we make this thread safe?

Reply:
Actually, this not needed, because all code will be run from the CoreData's writableContext, so it is already thread-safe